### ISSUE TITLE
Rename files and folders

### ISSUE THREAD

**Issue by fabri9532** [rikyoz/bit7z#43]
Is there a method to rename files and folders inside the compressed file?
If it doesn't exist, can it be implemented?

Fabrizio.

**Comment by rikyoz** [comment]
Hi Fabrizio!
At the moment, there is no such method.
It should be possible to implement such a feature, however it may require quite a few changes in both the external API and internal code of bit7z. 
Hopefully, I'll be able to implement it in the next stable version.
Riccardo

**Comment by fabri9532** [comment]
Thank you! Riccardo.

Fabrizio.

**Comment by msghera** [comment]
Is this thread still open for contribution ?

**Comment by rikyoz** [comment]
Hi! Yeah, sure!
I'm still working on implementing this feature; unfortunately, it requires a lot of changes in the internal code of bit7z.
During these last months, I've been busy with my main job, but now I'm going back to work on bit7z.
I hope to finish this and the other features asap.
Anyway, I'm open to any helpful contribution!

**Comment by msghera** [comment]
If the task is huge, you can share or I am find new areas for contribution.

**Comment by rikyoz** [comment]
Implemented in [v4.0.0](https://github.com/rikyoz/bit7z/releases/tag/v4.0.0).

Now the `BitArchiveEditor` can be used for renaming/deleting files and folders inside archives.

Example:
```cpp
BitArchiveEditor editor{ lib, "path/to/archive.7z", BitFormat::SevenZip };

// Set compressed_file1.txt to be renamed as compressed_file.txt.
editor.renameItem("compressed_file1.txt", "compressed_file.txt"); 

// Set folder/compressed_file2.txt to be deleted from the archive.
editor.deleteItem("folder/compressed_file2.txt");

// Applying the requested changes
editor.applyChanges();
```
